src/inmanta/util/dict_path.py

```diff
@@ -271,8 +271,23 @@ def get_path_sections(self) -> Sequence["WildDictPath"]:
     def parse(cls: type[TWDP], inp: str) -> Optional[TWDP]:
         pass
 
-    def _validate_container(self, container: object) -> TypeGuard[dict[object, object]]:
-        return isinstance(container, dict)
+    def _validate_container(
+        self, container: object, set: bool = False, remove: bool = False
+    ) -> TypeGuard[dict[object, object]]:
+        """Validate that the container supports the required mapping protocol operations
+
+        TODO: fix the typeguard. Probably by providing multiple methods for validation!
```

Comment:
There is actually a ticket to handle this. So we can remove the todo I guess. Currently the typeguard is no longer correct but we can split this method into readonly and writeable

Reply:
added reference to the ticket